lib/middleware/session_safe_logging.rb

```diff
@@ -8,15 +8,28 @@ def initialize(app)
 
     def call(env)
       session = env['rack.session']
+      ::Logging.mdc['remote_ip'] = env['action_dispatch.remote_ip'].try(:to_s)
+      ::Logging.mdc['request'] = env['action_dispatch.request_id']
 
-      # Store a UUID in the session to be used as an alternative to the session
-      # ID itself in logs, which might be hijacked.
-      session['session_safe'] ||= SecureRandom.hex(32)
-      ::Logging.mdc['session_safe'] = session['session_safe']
+      if env["rack.session"].id.present?
+        # use random token to prevent hijack.
+        session['logging_token'] ||= SecureRandom.uuid
+        ::Logging.mdc['session'] = session['logging_token']
+      else
+        # or store request token instead of session
+        ::Logging.mdc['session'] = env['action_dispatch.request_id']
+      end
 
       @app.call(env)
     ensure
-      ::Logging.mdc.delete('session_safe')
+      # variables set by middleware
+      ::Logging.mdc.delete('remote_ip')
+      ::Logging.mdc.delete('request')
```

Comment:
Since we do more here than just `session_safe` setup, we should probably rename the class to `LoggingMdcSetup` or something like that.

Reply:
Done, named it `LoggingContext` then.